### ISSUE TITLE
Fix UT failures caused by SQLite concurrency support

### ISSUE THREAD

**Issue by qiyuangong** [Intel-bigdata/SSM#1229]
We found that launching `mvn test` on SSM may ends up with database exceptions such as 'database is locked', 'database is busy' etc. That is because SQLite is not designed for large scale concurrency. It will locked the database file during updating. In our cases, multiple modules/threads may update metastore at the same time, which ends up with database exceptions. 

Note that SQLite is introduced as a temp metastore for Unit test in SSM.  Also, this issue will not effect any feature, but it has tripped us from a few weeks. :(

**Comment by qiyuangong** [comment]
Progress:

1. We have tried `Shared-memory` and `in-memory` features. They are not helpful in our case. 
2. We found that all UTs can be finished successfully on SSD, but failed on HDD. Because writing to SSD is much faster than HDD, which means that database/file lock interval will be much less. This difference make SQLite passes all UTs.

According to this progress, this issue cannot be solved immediately. I will switch to other issues and keep one eye on this issue.
